### PR TITLE
Emit action before directive be removed from DOM

### PR DESCRIPTION
**Pull request by k3nsei**
feat: emit action before directive destroy

chore: extend peer deps to include angular v17

Refs: #1418
Release-As: 16.1.0

### CODE REVIEW

File: projects/ng-in-viewport/src/lib/directives/in-viewport.directive.ts [hunk 1]
```diff
@@ -82,23 +82,31 @@ export class InViewportDirective implements AfterViewInit, OnDestroy {
     if (isPlatformBrowser(this.platformId)) {
       this.inViewportService.unregister(this.nativeElement, this.config);
     }
+
+    this.emit(undefined, true, false);
```
Comment (tinesoft) [inline projects/ng-in-viewport/src/lib/directives/in-viewport.directive.ts]: I believe this line should be moved right behind line  `this.inViewportService.unregister(this.nativeElement, this.config);` and wrapped by the `if (isPlatformBrowser(this.platformId)) { }` ... no point in emitting if on SSR...

File: projects/ng-in-viewport/src/lib/values/check-fn.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { Config } from './config';
 
 export interface InViewportCheckFnOptions {
```
Comment (tinesoft) [inline projects/ng-in-viewport/src/lib/values/check-fn.ts:5]: It would be nice adding  a small description of the fields here :-)

**Review by tinesoft — COMMENTED** [review]
Small comments added
Overall LGTM!

File: projects/ng-in-viewport/src/lib/directives/in-viewport.directive.ts (hunk 1 above)
Comment (k3nsei) [reply projects/ng-in-viewport/src/lib/directives/in-viewport.directive.ts]: True, I will put it inside that if statement.

There's no point having it before unregister call as it's just a cleanup function to dispose of the intersection observer.

**Pull request merged**